src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -58,12 +63,14 @@ public Course(@NotNull String name,
                 @NotNull List<Module> modules,
                 @NotNull List<Library> libraries,
                 @NotNull Map<String, String> requiredPlugins,
-                @NotNull Map<String, URL> resourceUrls) {
+                @NotNull Map<String, URL> resourceUrls,
+                @NotNull List<String> autoInstallComponentNames) {
     this.name = name;
     this.modules = modules;
     this.requiredPlugins = requiredPlugins;
     this.resourceUrls = resourceUrls;
     this.libraries = libraries;
+    this.autoInstallComponentNames = autoInstallComponentNames;
     components = Stream.concat(modules.stream(), libraries.stream())
```

Comment:
can this be extracted into a method and have some simple tests (for + and - cases e.g.)?